data_api/snapshot_convertor/src/main/scala/uk/ac/wellcome/platform/snapshot_convertor/services/ConvertorService.scala

```diff
@@ -2,17 +2,91 @@ package uk.ac.wellcome.platform.snapshot_convertor.services
 
 import javax.inject.Inject
 
+import akka.actor.ActorSystem
 import com.twitter.inject.Logging
-import uk.ac.wellcome.platform.snapshot_convertor.models.ConversionJob
+import uk.ac.wellcome.platform.snapshot_convertor.models.{
+  CompletedConversionJob,
+  ConversionJob
+}
 import uk.ac.wellcome.utils.GlobalExecutionContext.context
+import akka.http.scaladsl.model.Uri
+import akka.stream.ActorMaterializer
+import akka.stream.alpakka.s3.scaladsl.{MultipartUploadResult, S3Client}
+import akka.util.ByteString
+import uk.ac.wellcome.display.models.DisplayWork
+import uk.ac.wellcome.models.aws.AWSConfig
+import uk.ac.wellcome.models.{IdentifiedWork, WorksIncludes}
+import uk.ac.wellcome.utils.JsonUtil
+import uk.ac.wellcome.utils.JsonUtil._
 
 import scala.concurrent.Future
+import akka.stream.scaladsl.{Compression, Framing, Keep, Sink, Source}
+import com.twitter.finatra.json.FinatraObjectMapper
+import io.circe.Json
+import io.circe.parser.parse
+
+class ConvertorService @Inject()(actorSystem: ActorSystem,
+                                 awsConfig: AWSConfig,
+                                 s3Client: S3Client,
+                                 mapper: FinatraObjectMapper)
+    extends Logging {
 
-class ConvertorService @Inject()() extends Logging {
+  def runConversion(
+    conversionJob: ConversionJob): Future[CompletedConversionJob] = {
 
-  def runConversion(conversionJob: ConversionJob): Future[Unit] = {
     info(s"ConvertorService running $conversionJob")
 
-    Future.failed(new RuntimeException("Not implemented!"))
+    val objectKey = conversionJob.objectKey
+    val targetObjectKey = "target.txt.gz"
+
+    val (s3Source: Source[ByteString, _], _) = s3Client.download(
+      bucket = conversionJob.bucketName,
+      key = objectKey
+    )
+
+    val includes = WorksIncludes(
+      identifiers = true,
+      thumbnail = true,
+      items = true
+    )
+
+    val source = s3Source
```

Comment:
There should be a comment here explaining the format of the source file.

Reply:
I think maybe just a reference to the example in the test fixture could suffice? That's the thing that is guaranteed to be kept up to date by our build system and I don't think we want to maintain the same json in 2 places. But yeah, I agree that a reference to a sample output would be helpful.